pkg/remote/compression.go

```diff
@@ -31,26 +31,26 @@ const (
 )
 
 func SetRecvCompressor(ri rpcinfo.RPCInfo, compressorName string) {
-	if ri == nil {
+	if ri == nil || compressorName == "" {
 		return
 	}
-	rpcinfo.AsMutableEndpointInfo(ri.From()).SetTag("recv-compressor", compressorName)
+	rpcinfo.AsInvocationSetter(ri.Invocation()).SetExtra("recv-compressor", compressorName)
 }
 
 func SetSendCompressor(ri rpcinfo.RPCInfo, compressorName string) {
-	if ri == nil {
+	if ri == nil || compressorName == "" {
 		return
 	}
-	rpcinfo.AsMutableEndpointInfo(ri.From()).SetTag("send-compressor", compressorName)
+	rpcinfo.AsInvocationSetter(ri.Invocation()).SetExtra("send-compressor", compressorName)
```

Comment:
AsInvocationSetter can return nil, I think there is no need to wrap it, assert AsInvocationSetter directly